libarchive/archive_read_support_format_mtree.c

```diff
@@ -2141,7 +2149,14 @@ readline(struct archive_read *a, struct mtree *mtree, char **start,
 			} else if (u[0] == '\\') {
 				if (u[1] == '\n') {
 					/* Trim escaped newline. */
-					total_size -= 2;
+					/* Check if there's a '\r' before the '\n' for CRLF */
```

Comment:
This doesn't look right.  This code is concatenating lines when the line ends in `\` -- so on Unix, it's looking for `\` following by a NL character.  The CRLF analog would be to strip `\` followed by a CR-LF pair.  That would require something like:
```
} else if (u[0] == '\\') {
   if (u[1] == '\n') {
      /* Trim final backslash followed by NL */
      ...
   } else if (u[1] == '\r' && u[2] == '\n') {
     /* Trim final backslash followed by CRLF */
     ...
   }
```
```